cmd/storage-rest-client.go

```diff
@@ -298,11 +298,7 @@ func (client *storageRESTClient) DiskInfo(ctx context.Context) (info DiskInfo, e
 		client.diskInfoCache.Update = client.diskInfo
 	})
 	val, err := client.diskInfoCache.Get()
-	if err == nil {
-		info = val.(DiskInfo)
-	}
-
-	return info, err
+	return val.(DiskInfo), err
```

Comment:
this can panic if `val` is `nil` @vadmeste - such that 

```
if val != nil {
      return val.(DiskInfo), err
}
return info, err
```

Reply:
Good catch!